flink/v1.12/flink/src/main/java/com/netease/arctic/flink/write/hidden/AbstractHiddenLogWriter.java

```diff
@@ -139,7 +139,7 @@ public void initializeState(StateInitializationContext context) throws Exception
                 helper));
     int parallelism = getRuntimeContext().getNumberOfParallelSubtasks();
 
-    if (context.isRestored() && parallelismSame(parallelism)) {
+    if (context.isRestored() && parallelismSame(parallelism) && isValidCheckpointState()) {
```

Comment:
Does Flink 1.14/1.15 need to be modified as well? I don't see any changes in this PR